# RNA-seq Analysis of Acute and Chronic Periprosthetic Joint Infection (PJI) Isolates
This project investigates the transcriptional differences between acute and chronic Periprosthetic Joint Infection (PJI) isolates using RNA-seq data. The goal is to identify differentially expressed genes (DEGs) and long non-coding RNAs (lncRNAs) that may contribute to infection persistence, biofilm formation, and host‚Äìpathogen interactions.

## üóÇÔ∏è Project Structure
```
pji_analysis/
‚îú‚îÄ‚îÄ data/
‚îÇ   ‚îú‚îÄ‚îÄ raw_reads/                # Raw sequencing reads (FASTQ files)
‚îÇ   ‚îú‚îÄ‚îÄ trimmed_reads/            # Reads after quality trimming
‚îÇ   ‚îî‚îÄ‚îÄ mapping/                  # Alignment (mapping) outputs
‚îÇ
‚îú‚îÄ‚îÄ scripts/                      # All analysis scripts
‚îÇ
‚îú‚îÄ‚îÄ result/
‚îÇ   ‚îú‚îÄ‚îÄ gene_counts/              # Gene count files from featureCounts
‚îÇ   ‚îú‚îÄ‚îÄ star_logs/                # STAR alignment log files
‚îÇ   ‚îî‚îÄ‚îÄ QC_report/                # Quality control reports (FastQC, MultiQC, etc.)
‚îÇ
‚îî‚îÄ‚îÄ README.md                     # Project summary and analysis notes

refdata/
‚îú‚îÄ‚îÄ reference_genome.fa           # Reference genome
‚îî‚îÄ‚îÄ annotation.gtf                # Corresponding gene annotation file 

```
## üîß Pipeline Overview
This pipeline processes paired-end RNA-seq FASTQ files from Staphylococcus aureus isolates to identify transcriptional differences between acute and chronic phases of periprosthetic joint infection (PJI).

1. **Data Acquisition** 
2. **Quality Control** 
3. **Read Trimming** 
4. **Read Alignment** 
5. **Quantification** 
6. **Differential Expression Analysis** 
7. **Annotation** 
8. **Functional Enrichment**
9. **Visualization $ Interpretation**

## üõ†Ô∏è Tools & Dependencies

- Bash (Workflow scripting and automation)
- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [Fastp](https://github.com/OpenGene/fastp)
- [MultiQC](https://seqera.io/multiqc/)
- [STAR](https://github.com/alexdobin/STAR)
- [Samtools](http://www.htslib.org/)
- [Subread (featureCounts)](https://subread.sourceforge.net/featureCounts.html)
- [R 4.5.1](https://cran.ma.imperial.ac.uk/)
- [RTools 4.5](https://cran.r-project.org/bin/windows/Rtools/)
- [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)
- [Tidyverse](https://www.tidyverse.org/)
- [pheatmap](https://cran.r-project.org/web/packages/pheatmap/)
- [ComplexHeatmap](https://bioconductor.org/packages/release/bioc/html/ComplexHeatmap.html)
- [biomaRt](https://bioconductor.org/packages/release/bioc/html/biomaRt.html)
- [clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html)

---

## üöÄ How to Run
```
```

> ‚ö†Ô∏è Ensure that all required tools are installed and accessible in your system's `$PATH`.

## üì¶ Data Access

Due to GitHub file size restrictions, input FASTQ files and some intermediate outputs are **not included** in this repository.

To obtain the required data:

- Run `scripts/download_data.sh` which fetches all required files from public repositories, or  
- Manually download them using the accession numbers or URLs referenced inside the script

---
